src/gluonts/torch/util.py

```diff
@@ -139,3 +145,24 @@ def __init__(self, iterable):
 
     def __iter__(self):
         yield from self.iterable
+
+
+def unsqueeze_expand(a, dim, size):
+    a = a.unsqueeze(dim)
+    sizes = list(a.shape)
+    sizes[dim] = size
+    return a.expand(*sizes)
+
+
+def slice_along_dim(a, dim, slc):
+    idx = [slice(None)] * len(a.shape)
+    idx[dim] = slc
+    return a[idx]
+
+
+def repeat_along_dim(a, dim, repeats):
+    if repeats == 1:
+        return a
+    r = [1] * len(a.shape)
+    r[dim] = repeats
+    return a.repeat(*r)
```

Comment:
This is allocating, see https://pytorch.org/docs/stable/generated/torch.Tensor.repeat.html

Can probably get the same without allocations using `unsqueeze`, `expand`, and `view`.